tensorrt_llm/_torch/pyexecutor/sampler.py

```
@@ -247,22 +250,39 @@ def _meet_max_token_stop_criteria(self, request: LlmRequest):
+            try: 
+                new_words = tokenizer.decode(new_token,skip_special_tokens=False,clean_up_tokenization_spaces=False)
+            except Exception:
+                # If decode fails, fall back to token-based matching only
+                new_words = ""
             offset = 0
             for i, offset_end in enumerate(prefix_sum):
                 if i > 0:
                     offset = prefix_sum[i - 1]
                 stop_word = stop_words_list[offset:offset_end]
+                try:
+                    stop_text = tokenizer.decode(stop_word, skip_special_tokens=False, clean_up_tokenization_spaces=False)
```

Ruff (0.12.2)

Line too long (122 > 120)

(E501)